File: trunk/ivef-objc/Classes/IVEFBody.m

#import "IVEFBody.h"


@implementation IVEFBody

- (id) init {
    self = [super init];
    if (self != nil) {
        m_vesselDataPresent = false;
        m_vesselDatas = [[NSMutableArray alloc] init];
        m_loginRequestPresent = false;
        m_loginResponsePresent = false;
        m_pingPresent = false;
        m_pongPresent = false;
        m_serverStatusPresent = false;
        m_logoutPresent = false;
        m_serviceRequestPresent = false;
        m_dateFormatter = [[NSDateFormatter alloc] init];
        [m_dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS"];
    }
    return self;
}

-(void) addVesselData:(IVEFVesselData *) val {

    [m_vesselDatas addObject: val];
}

-(IVEFVesselData *) getVesselDataAt:(int) i {

    return [m_vesselDatas objectAtIndex: i];
}

-(int) countOfVesselDatas {

    return [m_vesselDatas count];
}

-(void) setLoginRequest:(IVEFLoginRequest *) val {

    m_loginRequestPresent = true;
    [m_loginRequest release];
    m_loginRequest = val;
    [m_loginRequest retain];
}

- (IVEFLoginRequest *) getLoginRequest {

    return m_loginRequest;
}

-(bool) hasLoginRequest {

    return m_loginRequestPresent;
}

-(void) setLoginResponse:(IVEFLoginResponse *) val {

    m_loginResponsePresent = true;
    [m_loginResponse release];
    m_loginResponse = val;
    [m_loginResponse retain];
}

- (IVEFLoginResponse *) getLoginResponse {

    return m_loginResponse;
}

-(bool) hasLoginResponse {

    return m_loginResponsePresent;
}

-(void) setPing:(IVEFPing *) val {

    m_pingPresent = true;
    [m_ping release];
    m_ping = val;
    [m_ping retain];
}

- (IVEFPing *) getPing {

    return m_ping;
}

-(bool) hasPing {

    return m_pingPresent;
}

-(void) setPong:(IVEFPong *) val {

    m_pongPresent = true;
    [m_pong release];
    m_pong = val;
    [m_pong retain];
}

- (IVEFPong *) getPong {

    return m_pong;
}

-(bool) hasPong {

    return m_pongPresent;
}

-(void) setServerStatus:(IVEFServerStatus *) val {

    m_serverStatusPresent = true;
    [m_serverStatus release];
    m_serverStatus = val;
    [m_serverStatus retain];
}

- (IVEFServerStatus *) getServerStatus {

    return m_serverStatus;
}

-(bool) hasServerStatus {

    return m_serverStatusPresent;
}

-(void) setLogout:(IVEFLogout *) val {

    m_logoutPresent = true;
    [m_logout release];
    m_logout = val;
    [m_logout retain];
}

- (IVEFLogout *) getLogout {

    return m_logout;
}

-(bool) hasLogout {

    return m_logoutPresent;
}

-(void) setServiceRequest:(IVEFServiceRequest *) val {

    m_serviceRequestPresent = true;
    [m_serviceRequest release];
    m_serviceRequest = val;
    [m_serviceRequest retain];
}

- (IVEFServiceRequest *) getServiceRequest {

    return m_serviceRequest;
}

-(bool) hasServiceRequest {

    return m_serviceRequestPresent;
}

-(void) setAttributes:(NSDictionary *)attributeDict {

        for (NSString *key in attributeDict) {
            if ([key isEqualToString: @"VesselData"]) {
                IVEFVesselData * val = [attributeDict objectForKey: key];
                [self addVesselData: val];
            }
            else if ([key isEqualToString:@"LoginRequest"]) {
                IVEFLoginRequest * val = [attributeDict objectForKey: key];
                [self setLoginRequest: val];
            }
            else if ([key isEqualToString:@"LoginResponse"]) {
                IVEFLoginResponse * val = [attributeDict objectForKey: key];
                [self setLoginResponse: val];
            }
            else if ([key isEqualToString:@"Ping"]) {
                IVEFPing * val = [attributeDict objectForKey: key];
                [self setPing: val];
            }
            else if ([key isEqualToString:@"Pong"]) {
                IVEFPong * val = [attributeDict objectForKey: key];
                [self setPong: val];
            }
            else if ([key isEqualToString:@"ServerStatus"]) {
                IVEFServerStatus * val = [attributeDict objectForKey: key];
                [self setServerStatus: val];
            }
            else if ([key isEqualToString:@"Logout"]) {
                IVEFLogout * val = [attributeDict objectForKey: key];
                [self setLogout: val];
            }
            else if ([key isEqualToString:@"ServiceRequest"]) {
                IVEFServiceRequest * val = [attributeDict objectForKey: key];
                [self setServiceRequest: val];
            }
        }
}

-(NSString *) XML {

    NSMutableString *xml = [NSMutableString stringWithString:@"<Body"];
    [xml appendString:@">\n"];
    for(int i=0; i < [m_vesselDatas count]; i++ ) {
        IVEFVesselData *attribute = [m_vesselDatas objectAtIndex:i];
        [xml appendString: [attribute XML] ];
    }
    if ( [self hasLoginRequest] ) {
        [xml appendString: [m_loginRequest XML] ];
    }
    if ( [self hasLoginResponse] ) {
        [xml appendString: [m_loginResponse XML] ];
    }
    if ( [self hasPing] ) {
        [xml appendString: [m_ping XML] ];
    }
    if ( [self hasPong] ) {
        [xml appendString: [m_pong XML] ];
    }
    if ( [self hasServerStatus] ) {
        [xml appendString: [m_serverStatus XML] ];
    }
    if ( [self hasLogout] ) {
        [xml appendString: [m_logout XML] ];
    }
    if ( [self hasServiceRequest] ) {
        [xml appendString: [m_serviceRequest XML] ];
    }
    [xml appendString: @"</Body>\n"];
    return xml;
}

-(NSString *) stringValueWithLead: (NSString *) lead {

    NSMutableString *str = [[[NSMutableString alloc] init] autorelease];
    [str setString: [lead stringByAppendingString:@"Body\n"]];
    for(int i=0; i < [m_vesselDatas count]; i++ ) {
        IVEFVesselData *attribute = [m_vesselDatas objectAtIndex:i];
        [str appendString: [attribute stringValueWithLead: [lead stringByAppendingString: @" "]] ];
    }
    if ( [self hasLoginRequest] ) {
        [str appendString: [m_loginRequest stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    }
    if ( [self hasLoginResponse] ) {
        [str appendString: [m_loginResponse stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    }
    if ( [self hasPing] ) {
        [str appendString: [m_ping stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    }
    if ( [self hasPong] ) {
        [str appendString: [m_pong stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    }
    if ( [self hasServerStatus] ) {
        [str appendString: [m_serverStatus stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    }
    if ( [self hasLogout] ) {
        [str appendString: [m_logout stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    }
    if ( [self hasServiceRequest] ) {
        [str appendString: [m_serviceRequest stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    }
    return str;
}


@end

File: branches/IVEF_0_1_RELEASE/ivef-qt/IVEFBody.cpp

#include "IVEFBody.h"

Body::Body() {

    m_vesselDataPresent = false;
    m_loginRequestPresent = false;
    m_loginResponsePresent = false;
    m_pingPresent = false;
    m_pongPresent = false;
    m_serverStatusPresent = false;
    m_logoutPresent = false;
    m_serviceRequestPresent = false;
}

Body::Body(const Body &val) : QObject() {

    m_vesselDataPresent = val.m_vesselDataPresent;
    m_vesselDatas = val.m_vesselDatas;
    m_loginRequestPresent = val.m_loginRequestPresent;
    m_loginRequest = val.m_loginRequest;
    m_loginResponsePresent = val.m_loginResponsePresent;
    m_loginResponse = val.m_loginResponse;
    m_pingPresent = val.m_pingPresent;
    m_ping = val.m_ping;
    m_pongPresent = val.m_pongPresent;
    m_pong = val.m_pong;
    m_serverStatusPresent = val.m_serverStatusPresent;
    m_serverStatus = val.m_serverStatus;
    m_logoutPresent = val.m_logoutPresent;
    m_logout = val.m_logout;
    m_serviceRequestPresent = val.m_serviceRequestPresent;
    m_serviceRequest = val.m_serviceRequest;
}

Body & Body::operator=(const Body &val) {

    m_vesselDataPresent = val.m_vesselDataPresent;
    m_vesselDatas = val.m_vesselDatas;
    m_loginRequestPresent = val.m_loginRequestPresent;
    m_loginRequest = val.m_loginRequest;
    m_loginResponsePresent = val.m_loginResponsePresent;
    m_loginResponse = val.m_loginResponse;
    m_pingPresent = val.m_pingPresent;
    m_ping = val.m_ping;
    m_pongPresent = val.m_pongPresent;
    m_pong = val.m_pong;
    m_serverStatusPresent = val.m_serverStatusPresent;
    m_serverStatus = val.m_serverStatus;
    m_logoutPresent = val.m_logoutPresent;
    m_logout = val.m_logout;
    m_serviceRequestPresent = val.m_serviceRequestPresent;
    m_serviceRequest = val.m_serviceRequest;
    return *this;
}

QString Body::encode( QString str) {

    str.replace('&', "&amp;");
    str.replace('<', "&lt;");
    str.replace('>', "&gt;");
    str.replace('"', "&quot;");
    return str;
}

void Body::addVesselData(VesselData val) {

    m_vesselDatas.append(val);
}

VesselData Body::getVesselDataAt(int i) const {

    return m_vesselDatas.at(i);
}

int Body::countOfVesselDatas() const {

    return m_vesselDatas.count();
}

void Body::setLoginRequest(LoginRequest val) {

    m_loginRequestPresent = true;
    m_loginRequest = val;
}

LoginRequest Body::getLoginRequest() const {

    return m_loginRequest;
}

bool Body::hasLoginRequest() {

    return m_loginRequestPresent;
}

void Body::setLoginResponse(LoginResponse val) {

    m_loginResponsePresent = true;
    m_loginResponse = val;
}

LoginResponse Body::getLoginResponse() const {

    return m_loginResponse;
}

bool Body::hasLoginResponse() {

    return m_loginResponsePresent;
}

void Body::setPing(Ping val) {

    m_pingPresent = true;
    m_ping = val;
}

Ping Body::getPing() const {

    return m_ping;
}

bool Body::hasPing() {

    return m_pingPresent;
}

void Body::setPong(Pong val) {

    m_pongPresent = true;
    m_pong = val;
}

Pong Body::getPong() const {

    return m_pong;
}

bool Body::hasPong() {

    return m_pongPresent;
}

void Body::setServerStatus(ServerStatus val) {

    m_serverStatusPresent = true;
    m_serverStatus = val;
}

ServerStatus Body::getServerStatus() const {

    return m_serverStatus;
}

bool Body::hasServerStatus() {

    return m_serverStatusPresent;
}

void Body::setLogout(Logout val) {

    m_logoutPresent = true;
    m_logout = val;
}

Logout Body::getLogout() const {

    return m_logout;
}

bool Body::hasLogout() {

    return m_logoutPresent;
}

void Body::setServiceRequest(ServiceRequest val) {

    m_serviceRequestPresent = true;
    m_serviceRequest = val;
}

ServiceRequest Body::getServiceRequest() const {

    return m_serviceRequest;
}

bool Body::hasServiceRequest() {

    return m_serviceRequestPresent;
}

QString Body::toXML() {

    QString xml = "<Body";
    xml.append(">\n");
    for(int i=0; i < m_vesselDatas.count(); i++ ) {
        VesselData attribute = m_vesselDatas.at(i);
        xml.append( attribute.toXML() );
    }
    if ( hasLoginRequest() ) {
        xml.append( m_loginRequest.toXML() );
    }
    if ( hasLoginResponse() ) {
        xml.append( m_loginResponse.toXML() );
    }
    if ( hasPing() ) {
        xml.append( m_ping.toXML() );
    }
    if ( hasPong() ) {
        xml.append( m_pong.toXML() );
    }
    if ( hasServerStatus() ) {
        xml.append( m_serverStatus.toXML() );
    }
    if ( hasLogout() ) {
        xml.append( m_logout.toXML() );
    }
    if ( hasServiceRequest() ) {
        xml.append( m_serviceRequest.toXML() );
    }
    xml.append( "</Body>\n");
    return xml;
}

QString Body::toString(QString lead) {

    QString str = lead + "Body\n";
    for(int i=0; i < m_vesselDatas.count(); i++ ) {
       VesselData attribute = m_vesselDatas.at(i);
       str.append( attribute.toString(lead + "    ") );
    }
    if ( hasLoginRequest() ) {
        str.append( m_loginRequest.toString(lead + "    ") );
    }
    if ( hasLoginResponse() ) {
        str.append( m_loginResponse.toString(lead + "    ") );
    }
    if ( hasPing() ) {
        str.append( m_ping.toString(lead + "    ") );
    }
    if ( hasPong() ) {
        str.append( m_pong.toString(lead + "    ") );
    }
    if ( hasServerStatus() ) {
        str.append( m_serverStatus.toString(lead + "    ") );
    }
    if ( hasLogout() ) {
        str.append( m_logout.toString(lead + "    ") );
    }
    if ( hasServiceRequest() ) {
        str.append( m_serviceRequest.toString(lead + "    ") );
    }
    return str;
}

